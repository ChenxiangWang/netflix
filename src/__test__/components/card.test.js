import React from "react";
import {
  render,
  screen,
  fireEvent,
  getAllByTestId,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

import Player from "../../components/player";
import Card from "../../components/card";
const category = "series";
const slideRows = [
  {
    title: "Comedies",
    data: [
      {
        maturity: "15",
        slug: "the-office",
        genre: "comedies",
        title: "The Office",
        description:
          "A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.",
        id: "59574941-9bbc-46d2-ad6b-c54af1019c93",
        docId: "1vTqgcZuDDXd9j8I1c34",
      },
    ],
  },
];

describe("<Card />", () => {
  const renderTestPage = () =>
    render(
      <Card.Group>
        {slideRows.map((slideItem) => {
          return (
            <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
              <Card.Title>{slideItem.title}</Card.Title>
              <Card.Entities>
                {slideItem.data.map((item) => (
                  <Card.Item key={item.docId} item={item} data-testid="trigger">
                    <Card.Image
                      src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                    />

                    {/* a popover cardition*/}
                    <Card.Meta data-testid="meta">
                      <Card.SubTitle>{item.title}</Card.SubTitle>
                      <Card.Text>{item.description}</Card.Text>
                    </Card.Meta>
                  </Card.Item>
                ))}
              </Card.Entities>
              <Card.Feature category={category} data-testid="feature">
                <Player>
                  <Player.Button />
                  <Player.Video src={"/videos/bunny.mp4"} />
                </Player>
              </Card.Feature>
            </Card>
          );
        })}
      </Card.Group>
    );

  it("test card's text", () => {
    const { container } = renderTestPage();
    // test for text
    expect(screen.getByText("Comedies")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  // it("test card's entity hover state", () => {
  //   const { container } = renderTestPage();
  //   const trigger = screen.getByTestId("trigger");
  //   const meta = screen.getByTestId("meta");

  //   expect(trigger).toBeInTheDocument();
  //   expect(meta).toHaveStyle({ display: "none" });
  //   userEvent.hover(trigger);
  //   fireEvent.mouseOver(trigger);
  //   expect(meta).toHaveStyle({ display: "block" });
  //   userEvent.unhover(trigger);
  //   expect(meta).not.toBeVisible();
  //   expect(container.firstChild).toMatchSnapshot();
  // });

  it("test card's click function", () => {
    const { container } = renderTestPage();
    const trigger = screen.getByTestId("trigger");
    expect(screen.queryByTestId("feature")).toBeFalsy();
    userEvent.click(trigger);
    expect(screen.getByTestId("feature")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
