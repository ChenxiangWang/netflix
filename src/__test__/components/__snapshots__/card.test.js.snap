// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> test card's click function 1`] = `
<div
  class="style__Group-sc-vuwtvm-2 eIsdkt"
>
  <div
    class="style__Container-sc-vuwtvm-1 icJcmq"
  >
    <p
      class="style__Title-sc-vuwtvm-0 eWTpGe"
    >
      Comedies
    </p>
    <div
      class="style__Entities-sc-vuwtvm-5 gyyxsV"
    >
      <div
        class="style__Item-sc-vuwtvm-8 faMJBz"
        data-testid="trigger"
      >
        <img
          class="style__Image-sc-vuwtvm-7 bjXbkp"
          src="/images/series/comedies/the-office/small.jpg"
        />
        <div
          class="style__Meta-sc-vuwtvm-6 YqkBR"
          data-testid="meta"
        >
          <p
            class="style__SubTitle-sc-vuwtvm-3 kUjijK"
          >
            The Office
          </p>
          <p
            class="style__Text-sc-vuwtvm-4 hiAxVH"
          >
            A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
          </p>
        </div>
      </div>
    </div>
    <div
      class="style__Feature-sc-vuwtvm-10 cjhxbR"
      data-testid="feature"
      src="/images/series/comedies/the-office/large.jpg"
    >
      <div
        class="style__Content-sc-vuwtvm-13 rqYEY"
      >
        <p
          class="style__Title-sc-vuwtvm-0 style__FeatureTitle-sc-vuwtvm-11 eWTpGe hJLsIY"
        >
          The Office
        </p>
        <p
          class="style__FeatureText-sc-vuwtvm-9 kZOGwR"
        >
          A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
        </p>
        <button
          class="style__FeatureClose-sc-vuwtvm-12 einrHM"
        >
          <img
            alt="Close"
            src="/images/icons/close.png"
          />
        </button>
        <div
          class="style__Group-sc-vuwtvm-2 gOiYXR"
        >
          <div
            class="style__Maturity-sc-vuwtvm-14 cMcAEG"
          >
            15
          </div>
          <p
            class="style__FeatureText-sc-vuwtvm-9 kZOGwR"
            font-weight="blod"
          >
            The Office
          </p>
        </div>
        <div
          class="style__Container-sc-1wkdspn-0"
        >
          <button
            class="style__Button-sc-1wkdspn-4 fYiLbt"
          >
            Play
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> test card's text 1`] = `
<div
  class="style__Group-sc-vuwtvm-2 eIsdkt"
>
  <div
    class="style__Container-sc-vuwtvm-1 icJcmq"
  >
    <p
      class="style__Title-sc-vuwtvm-0 eWTpGe"
    >
      Comedies
    </p>
    <div
      class="style__Entities-sc-vuwtvm-5 gyyxsV"
    >
      <div
        class="style__Item-sc-vuwtvm-8 faMJBz"
        data-testid="trigger"
      >
        <img
          class="style__Image-sc-vuwtvm-7 bjXbkp"
          src="/images/series/comedies/the-office/small.jpg"
        />
        <div
          class="style__Meta-sc-vuwtvm-6 YqkBR"
          data-testid="meta"
        >
          <p
            class="style__SubTitle-sc-vuwtvm-3 kUjijK"
          >
            The Office
          </p>
          <p
            class="style__Text-sc-vuwtvm-4 hiAxVH"
          >
            A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
